In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_csv("Compute_allocations_transformer.csv")
# byte_level = df[df['Patch size'] == 1]
byte_level = df[(df['FLOPs'] == 's') & (df['Patch size'] == 2)]
# byte_level['Global transformer parameters'].sort_values(ascending=False)
fig = px.histogram(byte_level['Global model parameters'])

fig.show()

In [5]:
byte_level[(byte_level['Global model parameters'] <= 3.2e7) & (byte_level['Global model parameters'] >= 2.8e7)]

,FLOPs,Patch size,Tokens,Encoder parameters,Global model parameters,Decoder parameters,Global model dimension,Encoder/Decoder dimension,Global model layers,Decoder layers
12,s,2,2.576442e+10,1229824.00,28335144.96,2459648.00,768,320,4,2
63,s,2,2.423669e+10,1229824.00,29761740.80,2459648.00,704,320,5,2
67,s,2,2.239401e+10,1770946.56,29761740.80,3541893.12,704,384,5,2
123,s,2,2.377637e+10,1229824.00,29515776.00,2459648.00,640,320,6,2
248,s,2,2.016762e+10,1770946.56,31877038.08,3541893.12,576,384,8,2
303,s,2,2.297110e+10,1229824.00,28335144.96,2459648.00,512,320,9,2
360,s,2,1.966084e+10,1770946.56,31483494.40,3541893.12,512,384,10,2
412,s,2,2.437497e+10,787087.36,29515776.00,2361262.08,640,256,6,3
501,s,2,2.045149e+10,1229824.00,31877038.08,3689472.00,576,320,8,3
547,s,2,2.352936e+10,787087.36,28335144.96,2361262.08,512,256,9,3


4e18 FLOPs
===

tokens, global model params, decoder params, global model dimension, encoder/decoder dimension, global model layers, decoder layers

Patch size = 2

4.094315e+10, 11247452,	1224212, 448, 256, 6, 2

1.993053e+10 31483494, 3689472, 512, 320, 10, 3 (30M run)

9.492046e+09, 60177695, 5508956, 704, 384, 13, 4

Patch size = 4

4.937337e+10, 21691515, 1836319, 576, 256, 7, 3

1.847138e+10, 60177695, 5508956, 704, 384, 13, 4

9.375604e+09, 120M, 12242125, 896, 512, 16, 5